keps/sig-storage/1979-object-storage-support/README.md

```diff
@@ -711,214 +645,324 @@ BucketAccess {
 
   Spec BucketAccessSpec {
     // BucketClaimName is the name of the BucketClaim.
+    // +required
     BucketClaimName string
 
-    // Protocol is the name of the Protocol 
-    // that this access credential is supposed to support
-    // If left empty, it will choose the protocol supported
-    // by the bucket. If the bucket supports multiple protocols,
-    // the end protocol is determined by the driver. 
-    // +optional
-    Protocol Protocol
-
-    // BucketAccessClassName is the name of the BucketAccessClass
+    // BucketAccessClassName is the name of the BucketAccessClass.
+    // +required
     BucketAccessClassName string
 
-    // CredentialsSecretName is the name of the secret that COSI should populate
-    // with the credentials. If a secret by this name already exists, then it is
-    // assumed that credentials have already been generated. It is not overridden.
-    // This secret is deleted when the BucketAccess is delted.
-    CredentialsSecretName string
-    
-    // ServiceAccountName is the name of the serviceAccount that COSI will map
-    // to the OSP service account when IAM styled authentication is specified
-	// +optional
-	ServiceAccountName string
+    // Protocol is the name of the Protocol that this access credential is expected to support.
+    // +required
+    Protocol Protocol
```

Comment:
If `Protocols` is important for BucketClaims/Buckets it stands to reason (to me) that `Protocol` is important for BucketAccess also.